edgexfoundry/go-mod-configuration
# build: Upgrade consul/api from 1.9.1 to 1.12.0 and fix failing unit test

Signed-off-by: Leonard Goodell <leonard.goodell@intel.com>







If your build fails due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-configuration/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

  <link to docs PR>

## Testing Instructions
Build edgex-go with this branch
Build ASC using this branch
Run secure docker stack using compose builder command:
```
make run ds-virtual
```
Stop Core Data and App Rules Engine containers
Using Consul UI delete the configuration for these two services
Run core data and ASC from command line 
```
sudo EDGEX_SECURITY_SECRET_STORE=true ./core-data -r -cp
sudo ./app-service-configurable -r -cp -p=rules-engine
Verify configs are loaded in Consul as normal (i.e. ACL token worked)
Change app-rules-engine log level to DEBUG from Consul
Verify DEBUG messages logged showing data flowing from Device Virtual

## New Dependency Instructions (If applicable)
